### DISCUSSION TITLE
Updating configs

### DISCUSSION THREAD

**Discussion by alexa22013** [emqx/emqx-operator#529] [Q&A]
Hi,
Is there anyway to update emqx config when using bootstrapconfig https://github.com/emqx/emqx-operator/blob/main/config/samples/emqx/v2alpha1/emqx-full.yaml. Updating using a config file apart from using the dashboard to change configs?

Thanks.

**Comment by Rory-Z** [comment]
Sorry, The bootstrap config can only be set at creation time or modified via the dashboard, there is no other way to do this at the moment.
This is because the bootstrap conf is mounted to emqx/emqx.conf, and the configuration modified during emqx runtime is stored in the data directory, which is merged with the etc directory once, and the data directory takes precedence over the etc directory, so even if we allow the bootstrap config to be modified after deployment, it will not necessarily take effect.

**Reply by alexa22013** [reply]
anyway to use a config reloader to updade the configs?

**Comment by alexa22013** [comment]
Will updating the cluster-override.conf file directly change the emqx config?

**Reply by Rory-Z** [reply]
The `data/cluster-override.conf`  holds the configuration changes made by the EMQX Dashboard, I'm not sure what the consequences of updating it directly would be.  @zhongwencool Any ideas ?

**Reply by alexa22013** [reply]
hey @zhongwencool Any ideas ?

**Reply by zhongwencool** [reply]
**Suggestion**: Don't modify data/*-override.conf manually.

Description of how the configuration file is loaded:
1. Configuration priority: `emqx.conf < ENV < cluster-overrride.conf`
2. The changes to the configuration via Dashboard will go into `cluster-override.conf`
3. Before node starts, `cluster-override.conf` will be copied from the core node in the cluster to this node. 

So if you modify `cluster-override.conf` manually, some changes may be overwritten by emqx.

**Reply by alexa22013** [reply]
So what exacly happens when cluster-override.conf file is modified by dashboard changes? Because when I change the file manually I see no changes reflected on the cluster or dashboard

**Reply by Rory-Z** [reply]
Sorry for delay, maybe need restart EMQX application after update `data/cluster-override.conf ` ? @zhongwencool

**Reply by zhongwencool** [reply]
> Sorry for delay, maybe need restart EMQX application after update `data/cluster-override.conf ` ? @zhongwencool

Need 
1. stop emqx cluster 
2. modify data/cluster-override.conf 
3. start emqx cluster
